server/src/completion/YmlCompletionItemsProvider.ts

```diff
@@ -37,4 +37,18 @@ export class YmlCompletionItemsProvider {
             })
             .filter((elem) => !!elem);
     }
+
+    /**
+     * Find the *first* completion item that has `entityName` as name.
+     *
+     * @param entityName The name of the entity searched for.
+     * @returns The first completion item that has the same label as `entityName`, `null` otherwise.
+     */
+    public getItemByLabel(entityName: string): AbstractYmlObject {
+        const sameNameElements = this.completions.filter((elem) => elem.label === entityName);
+        if (!sameNameElements || sameNameElements.length === 0) {
+            return null;
+        }
+        return sameNameElements.shift();
```

Comment:
Hum… Isn't there some way to avoid uselessly applying the filter on the whole array, since we end up only returning one element? I would even prefer a basic loop-based approach:

```
for(item in arr) {
    if(test) {
        return item;
    }
}

return null;
```

But maybe there are things in arrays like “findFirst” or whatever.

Reply:
Done. The method `find` did the trick.